src/activities/content/ContentFileEntity.js

```diff
@@ -1,15 +1,19 @@
-import { Actions, Rels } from '../../hypermedia-constants.js';
+import { Actions, Rels, Classes } from '../../hypermedia-constants.js';
 import { performSirenAction } from '../../es6/SirenAction.js';
 import ContentHelperFunctions from './ContentHelperFunctions.js';
 import { ContentWorkingCopyEntity } from './ContentWorkingCopyEntity.js';
 
+export const FILE_TYPES = {
+	html: 'html',
+};
+
 /**
- * ContentHtmlFileEntity class representation of a d2l content-htmlFile entity.
+ * ContentFileEntity class representation of a d2l content-file entity.

```

Comment:
This class still has an html file specific action `setHtmlFileHtmlContent`. If we get any other file entity in the future, they won't have this action, so calling this just won't do anything. As we add more file entities, there may be more methods that do not relate to the specific actions of the file entity.

We could instead have file subclasses for each different file type that would extend this `ContentFileEntity` (ex: remove `setHtmlFileHtmlContent` from `ContentFileEntity`, put it in a `ContentHtmlFileEntity` and have that extend `ContentFileEntity`). So we would need to start with `ContentFileEntity`, find out what type of file it is, then create the specific sub-entity. But I'm not sure if that would be worth it/overengineering (sorry for ramble).

Reply:
**Note**: we also don't have access to any other methods before calling the `checkout` action anyways, so having access to callable methods that don't do anything is already kind of present (unless we also added a sub entity with only the `checkout` action, which when called will return an entity which would have to be cast to `ContentFileEntity`, which again might be overkill).

Reply:
That's a good point. I'm inclined to go down the sub-class path like you mentioned so we can always be confident about what is available to us, which goes along with how hypermedia is supposed to work. It might be over engineering right now, but it's something we keep in mind as we continue to work on the file types.

Looking at things like this:
```
const action = this._entity.getActionByName(Actions.files.deleteFile);
		if (!action) {
			return;
		}
```
seems like something we should avoid doing, possibly. This is supposed to be the Siren representation of this entity and we should know or not if we actually have a certain action, I would think. Maybe this kind of guarding was just for safety though.

Reply:
I think we'll still need this guarding down the line, since a lot of actions won't be available if working copy is enabled and it hasn't been checked in yet. However, yes it would be weird if we have a pdf file and there is a `changeHtmlContent` method (or something) available. I'll pursue the sub-class path.

Reply:
makes sense!